test/arrayops.jl

```diff
@@ -362,6 +362,14 @@ a = [0,1,2,3,0,1,2,3]
 @test findprev(isodd, [2,4,5,3,9,2,0], 7) == 5
 @test findprev(isodd, [2,4,5,3,9,2,0], 2) == 0
 
+# find with general iterables
+s = "julia"
+@test find(s) == [1,2,3,4,5]
```

Comment:
This test conflicts directly with https://github.com/JuliaLang/julia/pull/16024. I also think that this is bizarre behavior:

```jl
julia> find("foo\0bar")
6-element Array{Int64,1}:
 1
 2
 3
 5
 6
 7
```

Sure, that's what falls out of the definition, but how is this a useful or meaningful operation?